crates/compilers/src/preprocessor/deps.rs

```diff
@@ -0,0 +1,275 @@
+            return;
+        }
+
+        self.referenced_contracts.insert(dependency.referenced_contract);
+        self.dependencies.push(dependency);
+    }
+}
+
+impl<'hir> Visit<'hir> for BytecodeDependencyCollector<'hir> {
+    type BreakValue = Never;
+
+    fn hir(&self) -> &'hir Hir<'hir> {
+        self.hir
+    }
+
+    fn visit_expr(&mut self, expr: &'hir Expr<'hir>) -> ControlFlow<Self::BreakValue> {
+        match &expr.kind {
+            ExprKind::New(ty) => {
+                if let TypeKind::Custom(item_id) = ty.kind {
+                    if let Some(contract_id) = item_id.as_contract() {
+                        let name_loc = SourceMapLocation::from_span(self.source_map, ty.span);
+                        let name = &self.src[name_loc.start..name_loc.end];
+                        // TODO: check if there's a better way to determine where constructor call
+                        // ends.
+                        let args_len = self.src[name_loc.end..].split_once(';').unwrap().0.len();
+                        self.collect_dependency(BytecodeDependency {
+                            kind: BytecodeDependencyKind::New(name.to_string(), args_len),
+                            loc: SourceMapLocation::from_span(
+                                self.source_map,
+                                Span::new(expr.span.lo(), expr.span.hi()),
+                            ),
+                            referenced_contract: contract_id.get(),
+                        });
+                    }
+                }
+            }
+            ExprKind::Member(member_expr, ident) => {
+                if ident.name.to_string() == "creationCode" {
+                    if let ExprKind::TypeCall(ty) = &member_expr.kind {
+                        if let TypeKind::Custom(contract_id) = &ty.kind {
+                            if let Some(contract_id) = contract_id.as_contract() {
+                                self.collect_dependency(BytecodeDependency {
+                                    kind: BytecodeDependencyKind::CreationCode,
+                                    loc: SourceMapLocation::from_span(self.source_map, expr.span),
+                                    referenced_contract: contract_id.get(),
+                                });
+                            }
+                        }
+                    }
+                }
+            }
+            _ => {}
+        }
+        self.walk_expr(expr)
+    }
+}
+
+/// Goes over all test/script files and replaces bytecode dependencies with cheatcode
+/// invocations.
+pub fn remove_bytecode_dependencies(
+    hir: &Hir<'_>,
+    deps: &PreprocessorDependencies,
+    data: &PreprocessorData,
+) -> Updates {
+    let mut updates = Updates::default();
+    for (contract_id, deps) in &deps.preprocessed_contracts {
+        let contract = Hir::contract(hir, ContractId::new(*contract_id));
+        let source = Hir::source(hir, contract.source);
```

Comment:
just use the methods `hir.source()`, same everywhere else where `Hir::` is used

Reply:
ah, yep, changed in https://github.com/foundry-rs/compilers/pull/252/commits/f4a109d53bfe97bba287335404c2f5cc20e7dbf6